app/liquid_tags/wikipedia_tag.rb

```diff
@@ -0,0 +1,79 @@
+class WikipediaTag < LiquidTagBase
+  PARTIAL = "liquids/wikipedia".freeze
+
+  def initialize(tag_name, input, tokens)
+    super
+    @data = get_data(input.strip)
+  end
+
+  def render(_context)
+    ActionController::Base.new.render_to_string(
+      partial: PARTIAL,
+      locals: {
+        title: @data[:title],
+        extract: @data[:extract],
+        url: @data[:url]
+      },
+    )
+  end
+
+  private
+
+  def valid_url?(input)
+    (input =~ /^https?\:\/\/([a-z-]+)\.wikipedia.org\/wiki\/([\S]+)$/)&.zero?
+  end
+
+  def get_data(input)
+    sanitized_url = ActionController::Base.helpers.strip_tags(input).strip
+    raise StandardError, "Invalid Wikipedia share URL" unless valid_url?(sanitized_url)
+
+    lang = URI.parse(sanitized_url).host.split(".").first
```

Comment:
We don't need to parse the URI repeatedly here, maybe just assign `uri = URI.parse(sanitized_url)` before doing the other 3 checks.